dtls/server-dtls-nonblocking.c

```diff
@@ -57,32 +51,95 @@ enum {
     TEST_ERROR_READY
 };
 
-int AwaitDGram(WOLFSSL_CTX* ctx)
+int main(int argc, char** argv)
 {
+    /* cont short for "continue?", Loc short for "location" */
+    int         cont = 0;
+    char        caCertLoc[] = "../certs/ca-cert.pem";
+    char        servCertLoc[] = "../certs/server-cert.pem";
+    char        servKeyLoc[] = "../certs/server-key.pem";
+    WOLFSSL_CTX* ctx;
+    /* Await Datagram variables */
     int     on = 1;
     int     res = 1;
     int     recvLen;                 /* length of string read */
     int     readWriteErr;
     int     listenfd = 0;            /* Initialize our socket */
     int     clientfd = 0;            /* client connection */
     int     len = sizeof(on);
-    int     cont;
     char    buff[MSGLEN];            /* string read from client */
     WOLFSSL* ssl = NULL;              /* Initialize ssl object */
     struct sockaddr_in servAddr;     /* our server's address */
     char    ack[] = "I hear you fashizzle\n";
+    /* DTLS set nonblocking flag */
+    int flags = fcntl(*(&listenfd), F_GETFL, 0);
+
+    /* "./config --enable-debug" and uncomment next line for debugging */
+    /* wolfSSL_Debugging_ON(); */
 
+    /* Initialize wolfSSL */
+    wolfSSL_Init();
+
+    /* Set ctx to DTLS 1.2 */
+    if ((ctx = wolfSSL_CTX_new(wolfDTLSv1_2_server_method())) == NULL) {
+        printf("wolfSSL_CTX_new error.\n");
+        return 1;
+    }
+    /* Load CA certificates */
+    if (wolfSSL_CTX_load_verify_locations(ctx,caCertLoc,0) !=
+            SSL_SUCCESS) {
+        printf("Error loading %s, please check the file.\n", caCertLoc);
+        return 1;
+    }
+    /* Load server certificates */
+    if (wolfSSL_CTX_use_certificate_file(ctx, servCertLoc, SSL_FILETYPE_PEM) != 
+            SSL_SUCCESS) {
+        printf("Error loading %s, please check the file.\n", servCertLoc);
+        return 1;
+    }
+    /* Load server Keys */
+    if (wolfSSL_CTX_use_PrivateKey_file(ctx, servKeyLoc,
+                SSL_FILETYPE_PEM) != SSL_SUCCESS) {
+        printf("Error loading %s, please check the file.\n", servKeyLoc);
+        return 1;
+    }
+
+    /* Await Datagram */
+    cont = 0;
     while (cleanup != 1) {
+        /* NonBlockingSSL_Accept variables */
+        int ret;
```

Comment:
These variables should be declared at the top of the function.  Some compilers will throw warnings about variables declared in the body of a function.